botocore/awsrequest.py

```diff
@@ -450,6 +471,92 @@ def prepare_body(self, data, files, json=None):
                 if 'Transfer-Encoding' in self.headers:
                     self.headers.pop('Transfer-Encoding')
 
+        if self.body and 'Content-Length' not in self.headers:
+            self.headers['Transfer-Encoding'] = 'chunked'
+
+
+class AWSResponse(object):
+
+    def __init__(self, url=None, status_code=None, headers=None, raw=None):
+        self.url = url
+        self.status_code = status_code
+        self.headers = HeadersDict(headers or {})
+        self.raw = raw
+
+        self._content = None
+
+    @property
+    def content(self):
+        """Content of the response, in bytes."""
+
+        if self._content is None:
+            # Read the contents.
+            try:
+                data_generator = self.raw.stream()
+            except AttributeError:
+                def stream():
+                    while True:
+                        chunk = self.raw.read(DEFAULT_CHUNK_SIZE)
+                        if not chunk:
+                            break
+                        yield chunk
+                data_generator = stream()
+            self._content = bytes().join(data_generator) or bytes()
+
+        return self._content
+
+    @property
+    def text(self):
+        if not self.content:
+            return str('')
+
+        return self.content.encode('utf-8')
```

Comment:
Shouldn't this take into account encodings provided via response headers?

Reply:
It could, requests had some fairly complex logic for trying to determine the content encoding. The only place the `text` property is used is in the `ContainerMetadataFetcher`. I think I might just remove this have do the decoding there.

Reply:
Switched this to just use content and decode.